### PR TITLE
Healthcheck

### PR DESCRIPTION
**Pull request by yashnevatia**
Adding a healthcheck endpoint for k8s [liveness and readiness checks](https://github.com/smartcontractkit/infra-k8s/blob/556fdaa778ed1f98aec661d90b0999ca66f2181a/projects/timelock-worker/files/shared/defaults.yaml#L12)
Ticket:
https://smartcontract-it.atlassian.net/browse/DEVEX-1436

### CODE REVIEW

File: cmd/start.go [hunk 1]
```diff
@@ -14,7 +17,7 @@ func startCommand() *cobra.Command {
 		startCmd = cobra.Command{
 			Use:   "start",
 			Short: "Starts the Timelock Worker daemon",
-			Run:   startHandler,
+			Run:   startHandlerAndServer,
```
Comment (fbac) [inline cmd/start.go]: The name `startHandler` is a convention on how to use cobra. I'd leave this as startHandler.

File: cmd/start.go [hunk 2]
```diff
@@ -38,7 +41,12 @@ func startCommand() *cobra.Command {
 	return &startCmd
 }
 
-func startHandler(cmd *cobra.Command, _ []string) {
+func startHandlerAndServer(cmd *cobra.Command, _ []string) {
```
Comment (fbac) [inline cmd/start.go]: Rename this function to startHandler.

File: cmd/start.go [hunk 3]
```diff
@@ -38,7 +41,12 @@ func startCommand() *cobra.Command {
...
+	startHandler(cmd)
+	startServer()
+}
+
+func startHandler(cmd *cobra.Command) {
```
Comment (fbac) [inline cmd/start.go]: This one can be renamed as startTimelock.

File: cmd/start.go [hunk 4]
```diff
@@ -84,3 +92,27 @@ func startHandler(cmd *cobra.Command, _ []string) {
...
 	logs.Info().Msg("shutting down timelock-worker")
 }
+
+func healthHandler(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprint(w, "OK")
+}
+
+// starts a http server, serving the healthz endpoint.
+func startServer() {
```
Comment (fbac) [inline cmd/start.go]: startServer is too generic, probably here we want something more specific like startHTTPServer.

File: cmd/start.go [hunk 5]
```diff
@@ -84,3 +92,27 @@ func startHandler(cmd *cobra.Command, _ []string) {
 
 	logs.Info().Msg("shutting down timelock-worker")
 }
+
+func healthHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (fbac) [inline cmd/start.go]: The health itself doesn't check any condition in timelock that requires it to be alive, the subscription could be dead and the web server still reporting ok.

File: cmd/start.go [hunk 6]
```diff
@@ -42,6 +49,11 @@ func startHandler(cmd *cobra.Command, _ []string) {
 	// Use this ctx as the base context.
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
+	go startHTTPHealthServer()
+	startTimelock(ctx, cmd)
```
Comment (yashnevatia) [inline cmd/start.go:53]: should startTimelock be called on another goroutine ?

**Review by fbac — APPROVED** [review]
LGTM.
As mentioned we'll refine this.

**Pull request merged**